src/php/Compiler/Analyzer/Environment/GlobalEnvironment.php

```diff
@@ -223,21 +230,33 @@ private function resolveWithAlias(Symbol $name, NodeEnvironmentInterface $env):
         return null;
     }
 
-    private function resolveWithoutAlias(Symbol $name, NodeEnvironmentInterface $env): ?GlobalVarNode
+    private function resolveWithoutAlias(Symbol $name, NodeEnvironmentInterface $env): ?AbstractNode
     {
         $ns = $this->getNs();
         if (isset($this->refers[$this->ns][$name->getName()])) {
             $ns = $this->refers[$this->ns][$name->getName()]->getName();
         }
 
-        // Try to resolve in current namespace
+        // Try to resolve interfaces in current namespace
+        if (isset($this->interfaces[$ns][$name->getName()])) {
+            return new PhpClassNameNode($env, Symbol::createForNamespace($ns, $name->getName()), $name->getStartLocation());
+        }
+
+        // Try to resolve in definitions current namespace
         $def = $this->getDefinition($ns, $name);
         if ($def) {
             return new GlobalVarNode($env, $ns, $name, $def, $name->getStartLocation());
         }
 
         // Try to resolve in phel.core namespace
         $ns = 'phel\core';
+
+        // Try to resolve interfaces in phel.core
+        if (isset($this->interfaces[$ns][$name->getName()])) {
+            return new PhpClassNameNode($env, Symbol::createForNamespace($ns, $name->getName()), $name->getStartLocation());
+        }
```

Comment:
This hardcoded `$ns` 🤔 
I understand why we're doing this, and it's fine to leave it, but I wanted to write a comment here for a future iteration.

Some immediate thoughts about this: apparently the logic of the `$ns = $this->getNs();` and `$ns = 'phel\core';` are the same, the only difference is the value inside `$ns`. Also there are some comments explaining what the code does. So my first idea would be to create a private function to encapsulate this logic and pass the value of `$ns` instead. Maybe something like this: 

```php
resolveNamespace(Symbol $name, NodeEnvironmentInterface $env, string $ns): ?AbstractNode
```

Reply:
Done 👉🏼  https://github.com/phel-lang/phel-lang/pull/394